### PR TITLE
Update to v9.6.0-IOFreeze

### LINKED ISSUE #4405: Update to EnergyPlus 9.6.0
Enhancement Request
--------------
Once v9.6.0-IOFreeze is released, we should update to it.

## Detailed Description
https://github.com/NREL/EnergyPlus/blob/develop/src/Transition/InputRulesFiles/Rules9-5-0-to-9-6-0.md

### Major changes
- [ ] `Space` and `SpaceList`:
  - [ ] `BuildingSurface:Detailed`
  - [ ] `Wall:Detailed`, `Wall:Exterior`, `Wall:Adiabatic`, `Wall:Underground`, `Wall:Interzone`
  - [ ] `RoofCeiling:Detailed`, `Roof`, `Ceiling:Adiabatic`, `Ceiling:Interzone`
  - [ ] `Floor:Detailed`, `Floor:GroundContact`, `Floor:Adiabatic`,`Floor:Interzone`
  - [ ] `InternalMass`
  - [ ] `ZoneProperty:UserViewFactors:BySurfaceName`
  - [ ] `People`
  - [ ] `Lights`
  - [ ] `ElectricEquipment`, `GasEquipment`, `HotWaterEquipment`, `SteamEquipment`, `OtherEquipment`, `ElectricEquipment:ITE:AirCooled`, `ZoneBaseboard:OutdoorTemperatureControlled`
  - [ ] `Daylighting:Controls`, `Daylighting:ReferencePoint`

### New additions
- [ ] `AirflowNetwork:MultiZone:SpecifiedFlowRate`
- [ ] `DesignSpecification:OutdoorAir:SpaceList`
  - [ ] `HVACTemplate*`, `Sizing:Zone`, `ZoneHVAC:IdealLoadsAirSystem`, `ZoneHVAC:HybridUnitaryHVAC`, `AirTerminal*`: add reference
  - [ ] `Controller:MechanicalVentilation`: modify a field and add references
- [ ] `CoilSystem:Cooling:Water`

### Minor stuff
- [x] `PerformancePrecisionTradeoffs`: added a field https://github.com/NREL/OpenStudio/commit/fd6d7caa8d74f79c3b3c551415a6e4c1ffd833bd
- [ ] `RunPeriod`: support new fields? https://github.com/NREL/OpenStudio/commit/3495764166855c0fe1e5975afe81a3fc622019c3
- [ ] `ComfortViewFactorAngles`: increase max extensible fields (not wrapped) https://github.com/NREL/OpenStudio/commit/3495764166855c0fe1e5975afe81a3fc622019c3
- [ ] `AirflowNetwork:MultiZone:ReferenceCrackConditions`: modify existing field https://github.com/NREL/OpenStudio/commit/3495764166855c0fe1e5975afe81a3fc622019c3
- [x] `Sizing:System`: modify field and add field (note: no ft/rt tests for this object) https://github.com/NREL/OpenStudio/commit/086f41b6a958f5216f6e3946235bea05ce2dbfc5
- [ ] `Coil:Cooling:Water:DetailedGeometry`: added a field (not wrapped) https://github.com/NREL/OpenStudio/commit/086f41b6a958f5216f6e3946235bea05ce2dbfc5
- [x] `AirConditioner:VariableRefrigerantFlow`: modify field names https://github.com/NREL/OpenStudio/commit/d969970b9e9701cb1eb68f5ade1a9ad057e7b54c
- [x] `AirLoopHVAC:OutdoorAirSystem`: remove a field 'Availability Manager List Name'
- [x] `HeatPump:PlantLoop:EIR:Cooling`, `HeatPump:PlantLoop:EIR:Heating`: we already fixed this in https://github.com/NREL/OpenStudio/pull/4368
- [ ] `GroundHeatExchanger:System`: add new field (not wrapped) https://github.com/NREL/OpenStudio/commit/3495764166855c0fe1e5975afe81a3fc622019c3
- [ ] `ComponentCost:LineItem`: modify a field (not wrapped) https://github.com/NREL/OpenStudio/commit/3495764166855c0fe1e5975afe81a3fc622019c3

### Other
- [x] Fix IddFixture test failures
- [x] Fix IdfFixture test failures
- [x] Fix IlluminanceMapFixture.IlluminanceMapPlotRefPts
- [x] Fix ModelFixture test failures
- [x] Fix EnergyPlusFixture test failures
- [x] Fix SqlFixture test failures (need to run `resources/utilities/SqlFile/update_all.sh`)
- [x] Fix epJSONFixture test failures
- [x] Fix RubyTest-IddObject_Test-ExtensibleIndex
- [x] Fix CLITest-IddObject_Test-ExtensibleIndex
- [x] Version translation (rename `update_3_2_1_to_3_2_2` to `update_3_2_1_to_3_3_0`?)

### PR DESCRIPTION
**Pull request by joseph-robertson**
Pull request overview
---------------------

 - Fixes #4405

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

**Comment by jmarrec** [comment]
@joseph-robertson I know why the epJSON tests are failing. There's a regression in the CLI (probably my fault when I implemented the std::filesystem PR...), it doesn't respect the output-directory parameter for convert-only. I'm addressing it in https://github.com/NREL/EnergyPlus/pull/9023

**Comment by tijcolem** [comment]
Last but not least leave it to windows to complain about this. 

```
D:\jenkins\openstudio\develop\src\utilities\geometry\Test\ThreeJS_GTest.cpp(57,5): error C2220: the following warning is treated as an error [D:\jenkins\openstudio\develop\build\src\utilities\openstudio_utilities_tests.vcxproj]

D:\jenkins\openstudio\develop\src\utilities\geometry\Test\ThreeJS_GTest.cpp(57,5): warning C4244: 'argument': conversion from 'unsigned __int64' to 'RawType', possible loss of data [D:\jenkins\openstudio\develop\build\src\utilities\openstudio_utilities_tests.vcxproj]

          with

          [

              RawType=double

```


Windows has warnings treated as errors otherwise this would be okay.  This is the offending line: https://github.com/antoine-galataud/OpenStudio/blob/feature/threejs-airloop-userdata/src/utilities/geometry/Test/ThreeJS_GTest.cpp#L57

Once that is fixed I think this can be merged in!

**Comment by tijcolem** [comment]
It appears the epJSONFixture unit test failures are failing due to energyplus cli not using the `--output-directory` option when converting idf to epjson. The unit tests expect the converted epjson files to exist in the defined `--output-directory` but they are in the cwd.  

unit test error message

```
C++ exception with description "Error during the E+ CLI call, epJSON wasn't created at /Users/tcoleman/git/OpenStudio/build/src/epjson/epjson_tests/CmplxGlz_SmOff_IntExtShading/eplus.epJSON
cmd = '/Users/tcoleman/git/OpenStudio/build/EnergyPlus-9.6.0-82391071e9-Darwin-macOS10.15-x86_64/energyplus-9.6.0 --output-directory /Users/tcoleman/git/OpenStudio/build/src/epjson/epjson_tests/CmplxGlz_SmOff_IntExtShading --convert-only /Users/tcoleman/git/OpenStudio/build/src/epjson/epjson_tests/CmplxGlz_SmOff_IntExtShading/eplus.idf'
```

I can confirm that running the the e+ cli call manually does the same and outputs to cwd vs  `--output-directory`

```bash
tcoleman-32966s:build tcoleman$ /Users/tcoleman/git/OpenStudio/build/EnergyPlus-9.6.0-82391071e9-Darwin-macOS10.15-x86_64/energyplus-9.6.0 --convert-only --output-directory "/Users/tcoleman/git/OpenStudio/build/src/epjson/epjson_tests/RefBldgLargeHotelNew2004_Chicago" /Users/tcoleman/git/OpenStudio/build/src/epjson/epjson_tests/RefBldgLargeHotelNew2004_Chicago/eplus.idf
EnergyPlus Starting
EnergyPlus, Version 9.6.0-82391071e9, YMD=2021.09.13 14:37
Converted input file format. Exiting.
EnergyPlus Run Time=00hr 00min  0.33sec
EnergyPlus Completed Successfully.
tcoleman-32966s:build tcoleman$ ls /Users/tcoleman/git/OpenStudio/build/src/epjson/epjson_tests/RefBldgLargeHotelNew2004_Chicago
eplus.idf    eplusout.end eplusout.err
tcoleman-32966s:build tcoleman$ /Users/tcoleman/git/OpenStudio/build/EnergyPlus-9.6.0-82391071e9-Darwin-macOS10.15-x86_64/energyplus-9.6.0 --convert-only --output-directory "/Users/tcoleman/git/OpenStudio/build/src/epjson/epjson_tests/RefBldgLargeHotelNew2004_Chicago" /Users/tcoleman/git/OpenStudio/build/src/epjson/epjson_tests/RefBldgLargeHotelNew2004_Chicago/eplus.idf
EnergyPlus Starting
EnergyPlus, Version 9.6.0-82391071e9, YMD=2021.09.13 14:42
Converted input file format. Exiting.
EnergyPlus Run Time=00hr 00min  0.35sec
EnergyPlus Completed Successfully.
tcoleman-32966s:build tcoleman$ ls /Users/tcoleman/git/OpenStudio/build/src/epjson/epjson_tests/RefBldgLargeHotelNew2004_Chicago
eplus.idf    eplusout.end eplusout.err
tcoleman-32966s:build tcoleman$ ls -l eplus.epJSON 
-rw-r--r--  1 tcoleman  NREL_NT\Domain Users  730933 Sep 13 14:42 eplus.epJSON
```

**Comment by jmarrec** [comment]
@tijcolem: https://github.com/NREL/EnergyPlus/pull/9023 will fix the epJSON stuff, cf my comment above from 2 weeks ago: https://github.com/NREL/OpenStudio/pull/4406#issuecomment-908300039

**Comment by jmarrec** [comment]
Have you run the reg tests and looked at EUI differences yet?

File: resources/model/OpenStudio.idd
Comment (jmarrec) [reply resources/model/OpenStudio.idd:19099]: For Sizing:System too

**Review by jmarrec — CHANGES_REQUESTED** [review]

File: resources/model/OpenStudio.idd
Comment (jmarrec) [reply resources/model/OpenStudio.idd:19099]: I added VT tests, VT rules, and had to make changes to the model file for ControllerMechanicalVentilation as well.

File: src/model/SizingSystem.cpp
Comment (jmarrec) [inline src/model/SizingSystem.cpp:1016]: @joseph-robertson A bunch of these are no longer present in the Ctor, while they should. At the very least the 

    setAllOutdoorAirinHeating("No");
    setAllOutdoorAirinCooling("No");

Please put them all back.

Comment (jmarrec) [reply src/model/SizingSystem.cpp:1016]: This is the major source of EUI deviations in the Regression tests so far

File: src/model/SizingSystem.cpp
Comment (jmarrec) [inline src/model/SizingSystem.cpp:961]: *  [ ] **TODO**

File: src/model/SizingSystem.cpp
Comment (jmarrec) [inline src/model/SizingSystem.cpp:696]: There's no point taking an optional double here, this is an antiquated way of doing things IMHO. Change to double only

**Review by jmarrec — CHANGES_REQUESTED** [review]

File: src/model/SizingSystem.cpp
Comment (joseph-robertson) [reply src/model/SizingSystem.cpp]: Ok so just to recap, this is what happened:
* I commented out set methods in the ctor that were just setting to default values (i.e., unnecessary calls to set methods)
* By commenting out `setAll` methods, we have the unintended consequence of using idd default values
* We don't want to use idd defaults because it causes EUI deviations, so we'll stick with non-default values

File: src/model/SizingSystem.cpp
Comment (joseph-robertson) [reply src/model/SizingSystem.cpp]: I'm not totally sure where/how to retrieve these value(s). I added some notes for where I found anything "occupant diversity" related in the eplusout sql file.

**Comment by tijcolem** [comment]
Reg results look good. 
https://github.com/NREL/OpenStudio-resources/runs/3723041013?check_suite_focus=true

Once the latest commit is done building I think this looks ready to drop in!

**Comment by tijcolem** [comment]
Ran reg tests:  https://github.com/NREL/OpenStudio-resources/runs/3723101539?check_suite_focus=true

No change in diff between this and https://github.com/NREL/OpenStudio/pull/4406 reg results so looks good to me.

File: src/model/SizingSystem.cpp
Comment (joseph-robertson) [reply src/model/SizingSystem.cpp]: https://github.com/NREL/OpenStudio/issues/4446

**Review by tijcolem — APPROVED** [review]
All looks good!

**Pull request merged**
